src/persist-client/src/inspect.rs

```diff
@@ -201,6 +205,67 @@ pub async fn fetch_state_diffs(
     Ok(live_states)
 }
 
+#[derive(Debug, serde::Serialize)]
+struct BatchPartOutput {
+    desc: Description<u64>,
+    updates: Vec<BatchPartUpdate>,
+}
+
+#[derive(Debug, serde::Serialize)]
+struct BatchPartUpdate {
+    k: String,
+    v: String,
+    t: u64,
+    d: i64,
+}
+
+#[derive(PartialOrd, Ord, PartialEq, Eq)]
+struct PrettyBytes<'a>(&'a [u8]);
+
+impl fmt::Debug for PrettyBytes<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match std::str::from_utf8(self.0) {
+            Ok(x) => fmt::Debug::fmt(x, f),
+            Err(_) => fmt::Debug::fmt(self.0, f),
+        }
+    }
+}
+
+/// Fetches the contents of a blob batch part
+pub async fn blob_batch_part(
+    blob_uri: &str,
+    shard_id: ShardId,
+    partial_key: String,
+) -> Result<impl serde::Serialize, anyhow::Error> {
+    let blob = BlobConfig::try_from(blob_uri).await?;
+    let blob = blob.clone().open().await?;
+
+    let key = PartialBatchKey(partial_key).complete(&shard_id);
+    let part = blob
+        .get(&*key)
+        .await
+        .expect("blob exists")
+        .expect("part exists");
+    let part = BlobTraceBatchPart::<u64>::decode(&part).expect("decodable");
+    let desc = part.desc.clone();
+
+    let mut encoded_part = EncodedPart::new(&*key, part.desc.clone(), part);
+    let mut out = BatchPartOutput {
+        desc,
+        updates: Vec::new(),
+    };
+    while let Some((k, v, t, d)) = encoded_part.next() {
+        out.updates.push(BatchPartUpdate {
```

Comment:
let's put in a flag to limit how many updates are included, with some reasonable default (20?). ideally we'd want to pull them from both the beginning and the end somehow